src/states/node.rs

```diff
@@ -662,7 +662,7 @@ impl Node {
             (ConnectionInfo(conn_info), ManagedNode(src_name), dst @ ManagedNode(_)) => {
                 self.handle_connection_info_from_node(conn_info, src_name, dst)
             }
-            (SectionUpdate { prefix, members }, Section(_), Section(_)) => {
+            (SectionUpdate { prefix, members }, Section(_), PrefixSection(_)) => {
```

Comment:
I think even a `SectionUpdate` could be sent _from_ a `PrefixSection`. It would ensure that even if they split immediately after sending, they all receive the `Ack` for it.
(Shouldn't make a big difference, though.)